tools/postgres/schema.sql

```diff
@@ -123,14 +123,14 @@ CREATE TABLE webknossos.datasets(
   sharingToken CHAR(256),
   logoUrl VARCHAR(2048),
   sortingKey TIMESTAMPTZ NOT NULL,
-  details JSONB,
+  metadata JSONB DEFAULT '[]',
```

Comment:
:thinking: My thought here was that it might be easier to work with at least an empty array instead of null. I added the null / None support to the code to ensure easier compatibility. This includes update calls not necessarily having to pass metadata (at least I hope that's the case, as I wanted to implement it that way).

But I can easily remove the default value from the column :) Or even go further and make metadata non-optional if you think this is preferable.

Reply:
I don’t really have a preference for default value versus nullable. But I don’t much like having both. Please pick what’s easier implementation-wise :) My guess is that removing the default value (and so have the default be NULL/None) is easier.

Reply:
Not is should be non nullable and with `DEFAULT '[]` for dataset and folders :).

The update route for datasets parses the metadata field as optional and used the already existing metadata in the backend query in case the user / client did not provide any metadata (untested) :see_no_evil: